### PR TITLE
Replace `ProjectPackages.withPlatformDefaults` with `withDefaults`

### PR DESCRIPTION
**Pull request by lemnik**
## Goal
Remove the confusing `withPlatformDefaults` and introduce a simpler `ProjectPackages.withDefaults` which includes the default Android & Dart packages.

## Testing
New unit tests to cover `ProjectPackages`

### CODE REVIEW

File: packages/bugsnag_flutter/lib/src/config.dart [hunk 1]
```diff
@@ -66,36 +66,41 @@ class ProjectPackages {
...
+  /// Combine the given set of `packageNames` with the [detected] packages.
+  /// This is useful when you are using `--split-debug-info` and only want to
+  /// specify your Dart packages in [Bugsnag.start].
   ///
   /// See also:
   /// - [detected]
   /// - [Android Configuration.projectPackages](https://docs.bugsnag.com/platforms/android/configuration-options/#projectpackages)
-  const ProjectPackages.withPlatformDefaults(Set<String> packageNames)
-      : this._internal(packageNames, true);
+  factory ProjectPackages.withDefaults(Set<String> packageNames) =>
```
Comment (gareththackeray) [inline packages/bugsnag_flutter/lib/src/config.dart:90]:
```suggestion
  factory ProjectPackages.withDefaults(Set<String> additionalPackageNames) =>
```

File: packages/bugsnag_flutter/lib/src/config.dart [hunk 2]
```diff
@@ -55,9 +55,9 @@ class EndpointConfiguration {
 /// packages you consider to be part of your app (as opposed to a library).
 ///
 /// [By default](ProjectPackages.detected) this is set according to the
-/// underlying platform (iOS or Android) and an attempt is made to discover
-/// the Dart package your application uses. This detection *will not work* if
-/// you build using `--split-debug-info`.
+/// underlying platform (e.g. the  Android) and an attempt is made to discover
```
Comment (tomlongridge) [inline packages/bugsnag_flutter/lib/src/config.dart:58]:
```suggestion
/// underlying platform (e.g. Android) and an attempt is made to discover
```

Comment (lemnik) [reply packages/bugsnag_flutter/lib/src/config.dart:58]: Now how did that get there? 🤔  😄

File: packages/bugsnag_flutter/lib/src/client.dart [hunk 3]
```diff
@@ -760,6 +764,72 @@ class Bugsnag extends Client with DelegateClient {
...
+  /// your application uses on Android.
+  ///
+  /// See also:
+  /// - [withDefaults]
+  const ProjectPackages.only(Set<String> packageNames)
+      : this._internal(packageNames, false);
+
+  /// Combine the given set of `packageNames` with the default packages.
+  /// This is useful when you are using `--split-debug-info` and only want to
+  /// specify your Dart packages in [Bugsnag.start].
```
Comment (gareththackeray) [inline packages/bugsnag_flutter/lib/src/client.dart:818]:
```suggestion
  /// This is useful when you are using `--split-debug-info` so need to
  /// specify your Dart packages in [Bugsnag.start].
```

File: packages/bugsnag_flutter/lib/src/client.dart [hunk 4]
```diff
@@ -760,6 +764,72 @@ class Bugsnag extends Client with DelegateClient {
...
+
+  const ProjectPackages._internal(
+    this._packageNames,
+    this._includeDefaults,
+  );
+
+  /// Specify the exact list of package names to consider as part of the
+  /// project. This should include packages from both Dart and any Java packages
+  /// your application uses on Android.
+  ///
```
Comment (gareththackeray) [inline packages/bugsnag_flutter/lib/src/client.dart:810]:
```suggestion
  /// your application uses on Android.
  /// This not include any defaults.
```

**Review by gareththackeray — COMMENTED** [review]
LGTM just a couple of docs suggestions

**Review by gareththackeray — APPROVED** [review]

**Review by nickdowell — APPROVED** [review]

**Pull request merged**